From 06431639074d784afecbac0732551079767af472 Mon Sep 17 00:00:00 2001
From: Loic Poulain <loic.poulain@intel.com>
Date: Wed, 1 Oct 2014 16:44:12 +0200
Subject: [PATCH 4/5] net: rfkill: gpio: OBDA8723 PM support

rtl8723 bluetooth chip (OBDA8723) has the following interface:
HOST <---UART---> CONTROLLER
HOST --- WAKE---> CONTROLLER (gpio res 0)
HOST ---ENABLE--> CONTROLLER (gpio res 1)
HOST <---WAKE---- CONTROLLER (gpio res 2)

Update the acpi desc to match this configuration.

Signed-off-by: Loic Poulain <loic.poulain@intel.com>
---
 net/rfkill/rfkill-gpio.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/net/rfkill/rfkill-gpio.c b/net/rfkill/rfkill-gpio.c
index 79b6918..a4c4826 100644
--- a/net/rfkill/rfkill-gpio.c
+++ b/net/rfkill/rfkill-gpio.c
@@ -313,11 +313,19 @@ static struct rfkill_gpio_desc acpi_default_gps = {
 	.host_wake_idx = -1,
 };
 
+static struct rfkill_gpio_desc acpi_bluetooth_wake = {
+	.type = RFKILL_TYPE_BLUETOOTH,
+	.reset_idx = -1,
+	.shutdown_idx = 1,
+	.wake_idx = 0,
+	.host_wake_idx = 2,
+};
+
 static const struct acpi_device_id rfkill_acpi_match[] = {
 	{ "BCM2E1A", (kernel_ulong_t)&acpi_default_bluetooth },
 	{ "BCM2E39", (kernel_ulong_t)&acpi_default_bluetooth },
 	{ "BCM2E3D", (kernel_ulong_t)&acpi_default_bluetooth },
-	{ "OBDA8723", (kernel_ulong_t)&acpi_default_bluetooth },
+	{ "OBDA8723", (kernel_ulong_t)&acpi_bluetooth_wake },
 	{ "BCM4752", (kernel_ulong_t)&acpi_default_gps },
 	{ "LNV4752", (kernel_ulong_t)&acpi_default_gps },
 	{ "BCM4752E", (kernel_ulong_t)&acpi_default_gps },
-- 
1.8.3.2

